src/actions/app.js

```diff
@@ -28,10 +30,31 @@ export const toggleEditState = () => ({
   type: TOGGLE_EDITING_STATE,
 });
 
-export const saveAnnotation = (annotationData) => ({
-  type: SAVE_ANNOTATION,
-  payload: annotationData,
-});
+export const saveAnnotation = (annotationData, id = null) => {
+  return (dispatch, getState) => {
+    dispatch({
+      type: SAVE_ANNOTATION,
+      payload: annotationData,
+    });
+    // If the snippet has not been saved, then saving an annotation shouldn't
+    // trigger a POST request to save the snippet (and its annotations)
+    if (!id) {
+      // Return a Promise object to make this action "thenable"
+      return Promise.resolve();
+    }
+    // Get the app state and stringify it
+    const stateString = JSON.stringify(getState().app);
+
+    // TODO: Dispatch action to show the snackbar with a relevant message based on the results
+    // Return the results of the POST request
+    return axios.post(`api/snippets/${id}`, { json: stateString, })
```

Comment:
Except there isn't, because the request will never affect the state being updated because it's just a _side effect_. If I had updated the state only if the request was successful, then yes, that is coupling, and the request is no longer a side effect, but that isn't the case.